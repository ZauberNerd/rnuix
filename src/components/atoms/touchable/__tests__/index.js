// @flow
jest.unmock('../');

import * as React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Touchable from '../';

const shallow = new ShallowRenderer();
const onPress = jest.fn();

describe('Touchable', () => {
    it('renders a View and no Touchable* if no onPress is provided', () => {
        const tree = shallow.render(<Touchable />);
        expect(tree).toMatchSnapshot();
    });

    it('renders a TouchableOpacity by default if onPress is provided', () => {
        const tree = shallow.render(<Touchable onPress={onPress} />);
        expect(tree).toMatchSnapshot();
    });

    it('renders a TouchableOpacity by default if onLongPress is provided', () => {
        const tree = shallow.render(<Touchable onLongPress={onPress} />);
        expect(tree).toMatchSnapshot();
    });

    it('renders a View if onPress and disabled are provided', () => {
        const tree = shallow.render(<Touchable onPress={onPress} disabled />);
        expect(tree).toMatchSnapshot();
    });

    it('handles onPress callback', () => {
        const tree = shallow.render(<Touchable onPress={onPress} />);
        tree.props.onPress();
        expect(onPress).toHaveBeenCalled();
    });
});
