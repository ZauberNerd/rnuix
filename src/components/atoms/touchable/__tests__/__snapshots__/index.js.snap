// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Touchable renders a TouchableOpacity by default if onLongPress is provided 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onLongPress={[Function]}
  onPress={undefined}
/>
`;

exports[`Touchable renders a TouchableOpacity by default if onPress is provided 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onLongPress={undefined}
  onPress={[Function]}
/>
`;

exports[`Touchable renders a View and no Touchable* if no onPress is provided 1`] = `
<View
  onLongPress={undefined}
  onPress={undefined}
/>
`;

exports[`Touchable renders a View if onPress and disabled are provided 1`] = `
<View
  onLongPress={undefined}
  onPress={[Function]}
/>
`;
